packages/mask/src/plugins/Savings/protocols/LDOProtocol.ts

```diff
@@ -0,0 +1,180 @@
+import BigNumber from 'bignumber.js'
+import { ProtocolCategory, SavingsNetwork, SavingsProtocol, ProtocolType } from '../types'
+
+export interface LidoContract {
+    type: EthereumTokenType
+    chainName: string
+    ldoContract: string
+    stEthContract: string
+}
+
+export const LidoContracts: { [key: number]: LidoContract } = {
+    [ChainId.Mainnet]: {
+        type: EthereumTokenType.ERC20,
+        chainName: 'Ethereum',
+        ldoContract: getSavingsConstants(ChainId.Mainnet).LIDO || ZERO_ADDRESS,
+        stEthContract: getSavingsConstants(ChainId.Mainnet).LIDO_STETH || ZERO_ADDRESS,
+    },
+    [ChainId.Gorli]: {
+        type: EthereumTokenType.ERC20,
+        chainName: 'Gorli',
+        ldoContract: getSavingsConstants(ChainId.Gorli).LIDO || ZERO_ADDRESS,
+        stEthContract: getSavingsConstants(ChainId.Gorli).LIDO_STETH || ZERO_ADDRESS,
+    },
+}
+
+export class LidoProtocol implements SavingsProtocol {
+    public category = ProtocolCategory.ETH
+    public type = ProtocolType.Lido
+    public name = 'Lido'
+    public image = 'lido'
+    public base = 'ETH'
+    public pair = 'stETH'
+    public decimals = 18
+    public apy = '0.00'
+    public balance = new BigNumber('0')
+    public availableNetworks: SavingsNetwork[] = [
+        {
+            chainId: ChainId.Mainnet,
+            chainName: 'Ethereum',
+            contractAddress: getSavingsConstants(ChainId.Mainnet).LIDO_STETH || ZERO_ADDRESS,
+        },
+        {
+            chainId: ChainId.Gorli,
+            chainName: 'Gorli',
+            contractAddress: getSavingsConstants(ChainId.Gorli).LIDO_STETH || ZERO_ADDRESS,
+        },
+    ]
+
+    public getFungibleTokenDetails(chainId: ChainId): FungibleTokenDetailed {
+        let contractAddress = ''
+
+        for (const network of this.availableNetworks) {
+            if (network.chainId === chainId) {
+                contractAddress = network.contractAddress
+            }
+        }
+
+        return {
+            type: 1,
+            chainId: chainId,
+            address: contractAddress,
+            symbol: 'stETH',
+            decimals: 18,
+            name: 'Liquid staked Ether 2.0',
+            logoURI: [
+                'https://static.debank.com/image/eth_token/logo_url/0xae7ab96520de3a18e5e111b5eaab095312d7fe84/f768023f77be7a2ea23c37f25b272048.png',
+                'https://tokens.1inch.io/0xae7ab96520de3a18e5e111b5eaab095312d7fe84.png',
+            ],
+        }
+    }
+
+    public async getApy() {
+        try {
+            const LidoAprUrl = 'https://cors.r2d2.to/?https://stake.lido.fi/api/steth-apr'
+            const response = await fetch(LidoAprUrl)
+            const apy = await response.text()
+            this.apy = apy
+            return apy
+        } catch (error) {
+            console.log('LDO `getApy()` error', error)
+            // Default APR is 5.30%
+            this.apy = '5.30'
+            return '5.30'
+        }
+    }
```

Comment:
According to the url, it should be APR instead.

You can learn the difference from https://www.americanexpress.com/en-us/banking/online-savings/articles/apy-vs-apr/